js/components/operatorSignIn/operatorSelection.tsx

```diff
@@ -25,17 +25,16 @@ export const OperatorSelection = ({
   const { result: nfcResult } = useNfc();
 
   useEffect(() => {
-    if (nfcResult.status === "success" && employees.status === "ok") {
-      const badge = findEmployeeByBadgeSerial(
-        employees.result,
-        nfcResult.data,
-      )?.badge;
-
-      if (badge) {
-        onOK({ number: badge, method: "nfc" });
-      } else {
-        onBadgeLookupError();
-      }
+    if (nfcResult.status === "success") {
+      const badgeSerial = nfcResult.data;
+      fetchEmployeeByBadgeSerial(badgeSerial).then(
```

Comment:
**suggestion:** I do want to flag that there's a potential race condition here where the `useEffect` dependencies change before this promise resolves. I think that because of the way the `useNfc` hook works in practice (it "locks in" the result of a successful scan) it's unlikely that it will cause an actual problem in practice but I haven't yet 100% convinced myself of that.

Reply:
Yeah, I'd been thinking about that too. I was leaning towards it being unlikely enough to pop up that I just didn't bother to put in the effort to fix it. Maybe I should leave a comment nearby acknowledging that the race condition exists, even if I don't fix it now?

Reply:
A comment sounds good! I think that the way that the `useNfc` hook works is the key part of the rationale here to mention.